critwall/src/main/java/org/opencadc/critwall/FileSyncJob.java

```diff
@@ -126,16 +131,32 @@ public FileSyncJob(URI artifactID, URI locatorServiceID, StorageAdapter storageA
         this.artifactDAO = artifactDAO;
     }
 
+    public void setOwner(final Subject owner) {
+        this.owner = owner;
+    }
+
     @Override
     public void run() {
         log.info("fileSyncJob start.");
+        if (this.owner == null) {
+            doSync();
+        } else {
+            Subject.doAs(this.owner, (PrivilegedAction<Void>) () -> {
+                doSync();
+                return null;
+            });
+        }
+    }
+
+    private void doSync() {
+        log.info("doSync start.");
         final List<URL> urlList;
         try {
-            urlList = getdownloadURLs(this.locatorService, this.artifactID);
+            urlList = getDownloadURLs(this.locatorService, this.artifactID);
         } catch (Exception e) {
             // fail - nothing can be done without the list, negotiation cannot be retried
             log.error("transfer negotiation failed.");
-            return;
+            throw new RuntimeException(e);
```

Comment:
Can you please put the causing exception in the log.error call? When this failure happens all detailed info is lost.